### PR TITLE
codeintel: Limit size of commit graph

### PR DESCRIPTION
**Pull request by efritz**
When calculating the set of uploads for each commit, we now only bring back the portion of the commit graph that has LSIF data and will be reasonably queried. This greatly reduces the number of commits that we need to calculate upload visibility for and will reduce the memory requirements (see https://github.com/sourcegraph/sourcegraph/issues/16030) when the bulk of the graph can be omitted.

### Notes

We will need to update the integration tests to deal with this new heuristic. The existing ones are broken because we do not upload indexes roughly in commit order (which should always occur in practice except for test uploads).

### CODE REVIEW

File: enterprise/cmd/frontend/internal/codeintel/background/commit_updater.go [hunk 1]
```diff
@@ -70,16 +70,51 @@ func (u *CommitUpdater) tryUpdate(ctx context.Context, repositoryID, dirtyToken
...
+	// expire old information for active repositories in a janitor process.
+	dump, ok, err := u.dbStore.OldestDumpForRepository(ctx, repositoryID)
 	if err != nil {
-		return errors.Wrap(err, "gitserver.Head")
+		return errors.Wrap(err, "store.OldestDumpForRepository")
 	}
 
-	graph, err := u.gitserverClient.CommitGraph(ctx, repositoryID, gitserver.CommitGraphOptions{})
+	var graph map[string][]string
+	if ok {
```
Comment (Strum355) [inline enterprise/cmd/frontend/internal/codeintel/background/commit_updater.go:89]: if `ok` == false, we pass a nil map to CalculateVisibleUploads. Is this ok or an impossible state?

Comment (efritz) [reply enterprise/cmd/frontend/internal/codeintel/background/commit_updater.go:89]: The users must iterate and not index the graph, so there's nothing to throw on here. Though it does read a bit strange, this is safe.

Comment (efritz) [reply enterprise/cmd/frontend/internal/codeintel/background/commit_updater.go:89]: (Updated after the merge conflict to be more obvious)

**Review by Strum355 — APPROVED** [review]
:ship:

**Pull request merged**